is_pto/preprocess_message.py

```diff
@@ -15,10 +15,10 @@ def clean_sqs_skype_formatting(message):
     cleaned_message = re.sub(r'\B@\w+', '', cleaned_message) #@mentions
     cleaned_message = re.sub(r'&.*?;', '', cleaned_message) #encoded strings
     cleaned_message = re.sub(r'^(\s)*$\n', '', cleaned_message) #emtpy lines
+    cleaned_message = re.sub(r'((?<![\\])["])((?:.(?!(?<![\\])\1))*.?)\1', '', cleaned_message) #whole quoted message
     cleaned_message = cleaned_message.replace('<legacyquote>', '')
     cleaned_message = cleaned_message.replace(',', ' ')
     cleaned_message = cleaned_message.replace('.', ' ')
-    cleaned_message = cleaned_message.replace('"', ' ')
```

Comment:
Why was this line removed?

Reply:
when I tried with Quotes "", it was replacing quotes. So had to remove it.

Reply:
@nilaya123 - the quote not being recognized here is not " or '. It is the markdown for a quoted message in Skype. 

When someone quotes you on Skype, it gets sent over the network with some markdown. That is what you need to fix. You can look at the kind of test Annapoorani and me ran on Wednesday (17-Mar-2021) on the PTO channel with Preedhi's PTO message. You will notice the PTO detector kept thinking that was a PTO message. That is the thing I want fixed.

Reply:
@qxf2 : Sure Arun, Here I am seeing the message is been quoted with markdown <>, so need to fix for it..Am I right?

Reply:
Yes, everything that has been marked down as a <quote>...</quote> needs to be stripped out @nilaya123

Reply:
@qxf2 : Sorry Arun to ask again, but I did not get the meaning of "marked down as a ... "??. As when I see quoted skype message, did'nt see ...
![is-pto_doubt](https://user-images.githubusercontent.com/30227077/112321608-2feec800-8cd6-11eb-8570-5ca9eed2506d.JPG)

Reply:
@nilaya123 - markdown is a way to encode styling into normal text. You have used it to format your text on your .md files or on Trello or on Jira. Like that Skype encodes some stuff like quoting someone else's message, emojis, etc using some markdown. You need to figure that out and then write a pre-processor that ignores it.

For example, in the screenshot above, Annapoorani's message looks like this when Skype marks it down and transmits it: 

<quote author="live:.cid.5fc-redacted-by-me" authorname="Preedhi Vivek" timestamp="1615950211" conversation="19:redacted-by-me@thread.skype" messageid="1615950210989" cuid="24-redacted-by-me"><legacyquote>[1615950211] Preedhi Vivek: </legacyquote><at id=" ">all</at> am unwell and would have to take a sick day. <legacyquote>\\n\\n&lt;&lt;&lt; </legacyquote></quote>Take care."

Reply:
@qxf2 : Arun, I have made changes. Is it ok? 
![is-pto-fix](https://user-images.githubusercontent.com/30227077/112461852-6126d100-8d86-11eb-987a-a82a5e1452c1.JPG)

Reply:
@nilaya123 - the test for this fix cannot be via the web frontend. When we post on Skype PTO channel it is not hitting the web application at all. So the screenshot you show is not very useful.

Can you add a better test as part of code itself? Like something in the tests directory that exercises your fix? When that test runs that test on master, it should fail. When it runs against your fix, it should pass.

Reply:
@qxf2 : Arun, I tried running this  simple test , and is working as expected. Passing on this branch and failing on master.
Is it ok? Also if it is ok, then can also add github actions to this repo.

Reply:
@nilaya123 - the thinking is correct this time around. But the tests do not make much sense. 

a) You should not be naming files 'unit_test_'
b) A unit test should be independent - you cannot expect the app to be running (like you expect the app on localhost:6464). What you have written is simply an API test. Can you please convert it to a true unit test?
c) I did not follow why preprocess message unit test has '<I am on PTO today>take care' as the example. Shouldn't you be testing with the quote string?
d) Have you verified that the tests for <quote blah ...> will fail against the current `main` branch?

Reply:
@qxf2 : Arun, I have made the changes as suggested.

1.Before I was masking all messages including inside quotes and after that too. But now I have done changes, as it is masking only messages in quotes. So only I have included those messages with quotes and string outside quotes, as need to test if some PTO message quoted should not get missed out.

2.Yes, I have verified the test on main branch, and they are failing.